src/App.tsx

```diff
@@ -14,11 +15,130 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const App: React.FC = () => {
+  const goodsList = [...goodsFromServer];
+  const [initialButtonVisible, start] = useState(true);
+  const [isReversed, reverse] = useState(false);
+  const [isSortedByLength, sortByLength] = useState(false);
+  const [isSortedByName, sortByName] = useState(false);
+  const [isReset, reset] = useState(false);
+  const [goodsLength, setGoodsLength] = useState('1');
+  const visibleGoods = goodsList.filter(good => good.length >= +goodsLength);
+
+  if (isSortedByLength) {
+    visibleGoods.sort((a, b) => a.length - b.length);
+  }
+
+  if (isSortedByName) {
+    visibleGoods.sort((a, b) => a.localeCompare(b));
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  if (isReset) {
+    reverse(false);
+    sortByLength(false);
+    sortByName(false);
+    reset(false);
+    setGoodsLength('1');
+  }
+
+  return (
+    <div className="App">
+      {!initialButtonVisible && (
+        <div className="App__container">
+          <h1 className="title has-text-centered">{`You can see ${visibleGoods.length} goods`}</h1>
+          <div className="App__flex-box container">
+            <GoodsList goods={visibleGoods} />
+          </div>
+
+          <div className="App__buttons buttons">
+            <button
+              type="button"
+              className="button is-primary is-small"
+              onClick={() => {
+                return (
+                  sortByName(true),
+                  sortByLength(false),
+                  reverse(false)
+                );
+              }}
```

Comment:
Look like you don't need return here, do you?
```suggestion
              onClick={() => {
                  sortByName(true),
                  sortByLength(false),
                  reverse(false)
              }}
```